ts/screens/modal/RegionServiceModal.tsx

```diff
@@ -0,0 +1,191 @@
+import CookieManager, { Cookie } from "@react-native-community/cookies";
+import { Body, Container, Content, Left, Text, View } from "native-base";
+import * as React from "react";
+import { Alert, ScrollView, TextInput } from "react-native";
+import { heightPercentageToDP } from "react-native-responsive-screen";
+import WebView, { WebViewMessageEvent } from "react-native-webview";
+import I18n from "../../i18n";
+import ButtonDefaultOpacity from "../../components/ButtonDefaultOpacity";
+import { Monospace } from "../../components/core/typography/Monospace";
+import ActivityIndicator from "../../components/ui/ActivityIndicator";
+import AppHeader from "../../components/ui/AppHeader";
+import IconFont from "../../components/ui/IconFont";
+import customVariables from "../../theme/variables";
+import { getCurrentLocale } from "../../utils/locale";
+import { showToast } from "../../utils/showToast";
+import {
+  APP_EVENT_HANDLER,
+  AVOID_ZOOM_JS,
+  closeInjectedScript
+} from "../../utils/webview";
+
+type Props = {
+  onModalClose: () => void;
+};
+
+const RegionServiceModal: React.FunctionComponent<Props> = (props: Props) => {
+  const [navigationURI, setNavigationUri] = React.useState("");
+  const [webMessage, setWebMessage] = React.useState("");
+  const [text, setText] = React.useState("");
+  const [loading, setLoading] = React.useState(false);
+  const [success, setSuccess] = React.useState(false);
+  const [error, setError] = React.useState(false);
+
+  React.useEffect(() => {
+    const cookie: Cookie = {
+      name: "test",
```

Comment:
I guess this is valid for your machine.
Could we pass a cookie to this component as a prop?